pyiso/nyiso.py

```diff
@@ -157,8 +158,9 @@ def get_any(self, label, parser, dates_list=None):
 
             # if fetch_csvs cannot get the individual days, it gets the whole month
             # Shortcut the loop if any call to fetch_csvs gets all dates in dates_list
+            date = pieces[-1].index[-1]
```

Comment:
For date ranges in `csv` that require a second iteration of the `while` loop, the line 161 changes the datatype of `date` from `datetime.date` to `datetime.datetime`. The next evaluation of line 151 will always return false since the `datetime` is not in `dates_list`. So this section of the method will end too early in that scenario.